### PR TITLE
[Feature][Connector V2] expose configurable options in Hudi

### PR DESCRIPTION
**Pull request by cason0126**
subtask of https://github.com/apache/incubator-seatunnel/issues/3337

### CODE REVIEW

File: seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/config/HudiSourceConfig.java [hunk 1]
```diff
@@ -17,18 +17,44 @@
...
+    public static final Option<String> CONF_FILES =
+            Options.key("conf.files")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("hudi conf files ");
+
+    public static final Option<String> USE_KERBEROS =
+            Options.key("use.kerberos")
+                    .stringType()
+                    .noDefaultValue()
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/config/HudiSourceConfig.java]: From the document I see `use.Kerberos` have a default value `false`.

File: seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+public class HudiSourceFactory implements TableSourceFactory {
+
+    @Override
+    public String factoryIdentifier() {
+        return "Hudi";
+    }
+
+    @Override
+    public OptionRule optionRule() {
+        return null;
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java]: Please return the OptionRule of this connector.

File: seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+public class HudiSourceFactory implements TableSourceFactory {
+
+    @Override
+    public String factoryIdentifier() {
+        return "Hudi";
+    }
+
+    @Override
+    public OptionRule optionRule() {
+        return OptionRule.builder().required(TABLE_PATH,TABLE_TYPE,CONF_FILES).build();
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java]: You lost the option options `use.kerberos` `kerberos.principal` `kerberos.principal.file`. And `kerberos.principal` `kerberos.principal.file` is required when `use.kerberos=true`. So you need add `conditional` . You can reference https://github.com/apache/incubator-seatunnel/pull/3320/files#diff-0626e040d3b207c066aee2e5e40f7faaf3d6dc1d01a87d097a21136b7b7092e3

Comment (cason0126) [reply seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java]: OK，I will FIX THEM SOON

Comment (cason0126) [reply seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java]: @EricJoy2048

File: seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/config/HudiSourceConfig.java [hunk 4]
```diff
@@ -17,18 +17,44 @@
...
+    public static final Option<String> CONF_FILES =
+            Options.key("conf.files")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("hudi conf files ");
+
+    public static final Option<Boolean> USE_KERBEROS =
+            Options.key("use.kerberos")
+                    .booleanType()
+                    .defaultValue(false)
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/config/HudiSourceConfig.java:45]: Should  we make `use_kerberos`  as an enum type option?

File: seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    @Override
+    public String factoryIdentifier() {
+        return "Hudi";
+    }
+
+    @Override
+    public OptionRule optionRule() {
+        return OptionRule.builder()
+                .required(TABLE_PATH,TABLE_TYPE,CONF_FILES)
+                .optional(USE_KERBEROS,KERBEROS_PRINCIPAL,KERBEROS_PRINCIPAL_FILE)
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java]:
```suggestion
                .optional(USE_KERBEROS)
```

File: seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/config/HudiSourceConfig.java (hunk 4 above)
Comment (EricJoy2048) [reply seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/config/HudiSourceConfig.java:45]: > Should we make `use_kerberos` as an enum type option?

Sorry, You don't need update it to enum type.

File: seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java (hunk 5 above)
Comment (cason0126) [reply seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java]: DONE

**Comment by EricJoy2048** [comment]
Please fix code style check

**Comment by cason0126** [comment]
I have fix code sytle check ; 
wating workflows approval ;
@EricJoy2048  THX!

File: seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+@AutoService(SeaTunnelSource.class)
+public class HudiSourceFactory implements TableSourceFactory {
+
+    @Override
+    public String factoryIdentifier() {
+        return "Hudi";
+    }
+
+    @Override
+    public OptionRule optionRule() {
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java:36]: Add UT for `optionRule()` please. We add some check rules for OptionRule builder, but need you add an UT to trigger it.

Comment (cason0126) [reply seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java:36]: OK, I Will Add it Soon .

Comment (cason0126) [reply seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java:36]: DONE

**Comment by EricJoy2048** [comment]
Please resolve conflicts.

**Comment by EricJoy2048** [comment]
fix code style please.

**Comment by cason0126** [comment]
OK。Sorry about that.
CS Setting does not including test source . 
I have fix it .

**Comment by cason0126** [comment]
How can I fix this problem ：Dead Links Fail 。
I do nothing but oneline to fix codestyle。 then，it happens。
@EricJoy2048

**Comment by cason0126** [comment]
fix dead link

**Comment by EricJoy2048** [comment]
You need merge dev branch code to fix the dead link.

**Comment by cason0126** [comment]
is there anything to fix?
@EricJoy2048

**Review by EricJoy2048 — APPROVED** [review]
LGTM

**Review by liugddx — APPROVED** [review]
+1

**Pull request merged**